contracts/bridges/polygon/FxBaseChildTunnel.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// IFxMessageProcessor represents interface to process message
+interface IFxMessageProcessor {
+    function processMessageFromRoot(
+        uint256 stateId,
+        address rootMessageSender,
+        bytes calldata data
+    ) external;
+}
+
+/**
+ * @notice Mock child tunnel contract to receive and send message from L2
+ */
+abstract contract FxBaseChildTunnel is IFxMessageProcessor {
```

Comment:
We don't need this. It's more of an example from Polygon.

Reply:
I still don't understand why there can't be a generic child that everyone shares to get the same interface as Optimism

Reply:
A couple arguments against it:

1. We need the bridge receiver contract to calculate `eta` using the L2 chain's block timestamp (instead of having it be an estimate that the 

2. there is no guarantee that relayed messages will be sent in a particular order (or that they will be received at all). with the bridge receiver, we're able to send a bundle of transactions to be queued that will all arrive at the same time. if we were relying on a generic receiver contract, we would need to send X transactions in a row and hope that they were all received and all received in the correct order, which won't always be the case

additionally, if we update the changes Kevin is suggesting below, the bridge receiver would also be responsible for ensuring that duplicate transactions were not queued and that the API for executeTransaction was simpler (taking a single transaction id)